backends/osx.py

```diff
@@ -72,7 +71,23 @@ def __dom_node_to_mote(self, port, dom_node):
         # Go up the DOM by a number of levels dependent on the macOS version. This will
         # This will be the <dict> element that appears when a device gets connected.
         if macos_ver == '10.15':
-            parent = dom_node.parentNode.parentNode.parentNode
+            parent = dom_node.parentNode
+
+            # We cannot predetermine the level of nesting, so we need to keep going up
+            # the DOM until we find the correct level. Unfortunately, we can only do this
+            # heuristically. We keep going up until we encounter a DOM <dict> that has a
+            # "USB Vendor Name"
+            valid = False
+            while not valid:
+                parent = parent.parentNode
+                child = parent.firstChild
```

Comment:
Do we not need to check if there is a parent in case a device is missing 'USB Vendor Name' ?